### PR TITLE
GH-594 : On Startup - Command Line Parameters execute Shell command

### LINKED ISSUE #631: Add AUTH support for the shell
As a developer, I'd like to add auth support for shell, so I can authenticate using user/pass credentials. This was originally addressed via spring-cloud/spring-cloud-dataflow#483, but it needs reworked and added back in.

### PR DESCRIPTION
**Pull request by markpollack**
* New parser that does not interpret command line options as shell commands

Other improvements

* Remove @ConditonalOnProperty disableInternalCommands
* Add @ImportResource("classpath*:/META-INF/spring/spring-shell-plugin.xml")
* Support --dataflow.uri, username, password command line options in ConfigCommands
* Introduce Boot based ShellProperties for historySize and commandLine
* Merge with master for oauth support in shell
* Polish configuration code of shell
* Update usage banner
* Update tests

Resolves #631 and #594

### CODE REVIEW

File: spring-cloud-dataflow-shell/src/main/resources/usage.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Data Flow Options:
+  --dataflow.url=<url>                  Port of the Data Flow Server [default: http://localhost:9393].
```
Comment (ericbottard) [inline spring-cloud-dataflow-shell/src/main/resources/usage.txt]: Not really "port", but rather "address" I'd say

File: spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java [hunk 2]
```diff
@@ -105,21 +103,22 @@ public static void startUp() throws InterruptedException, IOException {
...
 			}
 
-			SpringApplication application = new SpringApplicationBuilder(TestConfig.class).build();
+			SpringApplication application = new SpringApplicationBuilder(DataFlowServerConfiguration.class).build();
 
 			int randomPort = SocketUtils.findAvailableTcpPort();
+			String dataFlowUri = String.format("--dataflow.uri=http://localhost:%s", serverPort);
 			String dataSourceUrl = String.format("jdbc:h2:tcp://localhost:%s/mem:dataflow", randomPort);
 			applicationContext = application.run(
 					String.format("--server.port=%s", serverPort),
```
Comment (ericbottard) [inline spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java]: Is this still used then?

Comment (markpollack) [inline spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java]: Yes, this is the random http port of the server.

File: spring-cloud-dataflow-shell/src/main/resources/usage.txt (hunk 1 above)
Comment (ghillert) [inline spring-cloud-dataflow-shell/src/main/resources/usage.txt]: Will fix during merge. Also, changing `dataflow.user` to `dataflow.username`.

**Comment by ghillert** [comment]
Merged with commit 37d4863

**Pull request closed without merging**